modules/market/src/main/java/com/opengamma/strata/market/surface/DefaultSurfaceMetadataBuilder.java

```diff
@@ -0,0 +1,277 @@
+    this.xValueType = xValueType;
+    return this;
+  }
+
+  //-------------------------------------------------------------------------
+  /**
+   * Sets the y-value type, providing meaning to the y-values of the surface.
+   * <p>
+   * This type provides meaning to the y-values.
+   * <p>
+   * If using the builder, this defaults to {@link ValueType#UNKNOWN}.
+   * 
+   * @param yValueType  the y-value type
+   * @return this, for chaining
+   */
+  public DefaultSurfaceMetadataBuilder yValueType(ValueType yValueType) {
+    ArgChecker.notNull(yValueType, "yValueType");
+    this.yValueType = yValueType;
+    return this;
+  }
+
+  //-------------------------------------------------------------------------
+  /**
+   * Sets the z-value type, providing meaning to the z-values of the surface.
+   * <p>
+   * This type provides meaning to the z-values.
+   * <p>
+   * If using the builder, this defaults to {@link ValueType#UNKNOWN}.
+   * 
+   * @param zValueType  the z-value type
+   * @return this, for chaining
+   */
+  public DefaultSurfaceMetadataBuilder zValueType(ValueType zValueType) {
+    ArgChecker.notNull(zValueType, "zValueType");
+    this.zValueType = zValueType;
+    return this;
+  }
+
+  //-------------------------------------------------------------------------
+  /**
+   * Sets the day count.
+   * <p>
+   * This stores the day count in the additional information map using the
+   * key {@link SurfaceInfoType#DAY_COUNT}.
+   * <p>
+   * This is stored in the additional information map using {@code Map.put} semantics,
+   * removing the key if the day count is null.
+   * 
+   * @param dayCount  the day count, may be null
+   * @return this, for chaining
+   */
+  public DefaultSurfaceMetadataBuilder dayCount(DayCount dayCount) {
+    return addInfo(SurfaceInfoType.DAY_COUNT, dayCount);
+  }
+
+  //-------------------------------------------------------------------------
+  /**
+   * Adds a single piece of additional information.
+   * <p>
+   * This is stored in the additional information map using {@code Map.put} semantics,
+   * removing the key if the instance is null.
+   * 
+   * @param <T>  the type of the info
+   * @param type  the type to store under
+   * @param instance  the instance to store, may be null
+   * @return this, for chaining
+   */
+  public <T> DefaultSurfaceMetadataBuilder addInfo(SurfaceInfoType<T> type, T instance) {
+    ArgChecker.notNull(type, "type");
+    if (instance != null) {
+      this.info.put(type, instance);
+    } else {
+      this.info.remove(type);
+    }
+    return this;
+  }
+
+  /**
+   * Adds additional information.
+   * <p>
+   * This is stored in the additional information map using {@code Map.putAll} semantics
+   * 
+   * @param info  the information to add
+   * @return this, for chaining
+   */
+  public DefaultSurfaceMetadataBuilder addInfo(Map<SurfaceInfoType<?>, Object> info) {
+    ArgChecker.notNull(info, "infoMap");
+    this.info.putAll(info);
```

Comment:
This isn't type safe. If one of the values doesn't match the expected type of the surface info there will be a `ClassCastException` when the value is looked up. An exception should be thrown here if possible.